modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/ObjectEntryModelListener.java

```diff
@@ -101,9 +101,44 @@ public void onAfterRemove(ObjectEntry objectEntry)
 			throw new ModelListenerException(portalException);
 		}
 
-		_executeObjectActions(
-			ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE, objectEntry,
-			objectEntry);
+		ObjectEntry objectEntry1 = objectEntry;
+
+		try {
+			long userId = PrincipalThreadLocal.getUserId();
+
+			if (userId == 0) {
+				userId = objectEntry1.getUserId();
+			}
+
+			User user = _userLocalService.getUser(userId);
+
+			_executeObjectActions(
+				ObjectActionTriggerConstants.KEY_ON_AFTER_DELETE, objectEntry,
+				objectEntry1, user);
+
+			if (FeatureFlagManagerUtil.isEnabled("LPS-187142")) {
```

Comment:
Can you change the code back to the way it was? Brian prefers to fail first (the way it was in the method that you deleted).